includes/Core/Modules/Module_Sharing_Settings.php

```diff
@@ -0,0 +1,99 @@
+<?php
+/**
+ * Class Google\Site_Kit\Core\Modules\Module_Sharing_Settings
+ *
+ * @package   Google\Site_Kit\Core\Modules
+ * @copyright 2022 Google LLC
+ * @license   https://www.apache.org/licenses/LICENSE-2.0 Apache License 2.0
+ * @link      https://sitekit.withgoogle.com
+ */
+
+namespace Google\Site_Kit\Core\Modules;
+
+use Google\Site_Kit\Core\Storage\Setting;
+
+/**
+ * Base class for module sharing settings.
+ *
+ * @since n.e.x.t
+ * @access private
+ * @ignore
+ */
+class Module_Sharing_Settings extends Setting {
+
+	const OPTION = 'googlesitekit_dashboard_sharing';
+
+	/**
+	 * Gets the default value.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return array
+	 */
+	protected function get_default() {
+		return array();
+	}
+
+	/**
+	 * Gets the expected value type.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return string The type name.
+	 */
+	protected function get_type() {
+		return 'object';
+	}
+
+	/**
+	 * Gets the callback for sanitizing the setting's value before saving.
+	 *
+	 * @since n.e.x.t
+	 *
+	 * @return callable|null
+	 */
+	protected function get_sanitize_callback() {
+		return function( $option ) {
+			if ( is_array( $option ) ) {
+				foreach ( $option as &$sharing_settings ) {
+					if ( isset( $sharing_settings['sharedRoles'] ) ) {
+						$sharing_settings['sharedRoles'] = $this->filter_valid_roles( $sharing_settings['sharedRoles'] );
+					}
+					if ( isset( $sharing_settings['management'] ) ) {
+						$sharing_settings['management'] = (string) $sharing_settings['management'];
```

Comment:
AC says:

> `management` can only be one of the following: `owner` or `all_admins` (default: `owner`)

Let's make sure that `$sharing_settings['management']` is either `owner` or `all_admins` with a fallback to `owner`.

Reply:
I had originally implemented this but when I had a call with @aaemnnosttv on this and another issue, he suggested I either typecast or remove invalid elements and not modify elements or set default elements in the `sanitize_callback()`.

Reply:
@jimmymadon correct, sanitizing should be neutral or subtractive, not additive. I think the other part of that that we talked about was potentially adding a method for getting the settings for a specific module, which is where any missing parts would be filled in with defaults. This method could be internal for now and called in `get()` as defaults should be applied in the return value as well.

Reply:
@aaemnnosttv @eugene-manuilov So should I create this new method to get settings per module and override `Setting::get()` in `Module_Sharing_Settings` to use this new method?

Reply:
@jimmymadon, yes.

Reply:
@eugene-manuilov @aaemnnosttv Creating a new method to individually get settings per module required me to call `get()` anyways. So for now, I have updated the PR to only override the `Setting::get()` method that fills in a default value for the `management` key. This method can then be used to fetch settings for an individual module in the future.

Reply:
@jimmymadon left a comment on the method there, but please add the default for `sharedRoles` as well.